components/brave_wallet/browser/bitcoin/bitcoin_discover_account_task.h

```diff
@@ -71,7 +73,7 @@ class DiscoverAccountTaskBase {
   uint32_t active_requests_ = 0;
   // Indexed by 0 and 1 for receive and change addresses discovery states
   // respectively.
-  State states_[2];
+  std::array<State, 2> states_;
```

Comment:
This works, but IMO it hides the original problem here: that we are using some arbitrary index to access two static states. I've went looking around and it seems the issue starts here:

```
struct BitcoinKeyId {
  uint32 change;
  uint32 index;
};
```

So we are storing this index as a number, when in fact there is only two options, namely, `kBitcoinReceiveIndex` and `kBitcoinChangeIndex`.

I think we should consider changing this into two fields, and actually turning `BitcoinKeyId.index` into `BitcoinKeyId.type`, the type being an enum. What do you think?

Reply:
That's a good point and I believe there is a TODO somewhere to switch to enum or a bool there. But that would be too much for this PR.